Cyborg/SwiftUIView.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  Copyright (c) 2019. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  
+
+#if canImport(SwiftUI)
+
+import SwiftUI
+
+/// A `VectorDrawableView` for SwiftUI.
+///
+/// # Experimental
+/// This feature is experimental and may change.
+/// Current areas where this API may be deficient are:
+///
+/// 1. The drawable must be fully parsed before the view's init is called, which isn't in the spirit of SwiftUI's views being
+/// easy to construct.
+///
+/// 2. Handling of resources is a bit hacky. See below for details.
+///
+/// 3. Further testing of how the view handles sizing is required.
+///
+/// # Usage
+///
+/// You initialize a `VectorDrawableView` with
+/// `init(_ drawable:)`.
+///
+/// You can provide the `theme` and `resources` through
+/// `environment()`.
+///
+/// - Note: `resources` does not update after the view is first created,
+/// it's passed as an `Environment` var solely for convenience.
+@available(iOS 13.0, *)
+public struct VectorDrawableView: UIViewRepresentable {
+    
+    public let drawable: VectorDrawable
+    
+    @Environment(\.vectorDrawableTheme)
+    public var theme: ThemeProviding
+    
+    @Environment(\.vectorDrawableResources)
+    public var resources: ResourceProviding
+    
+    /// Initializer.
+    public init(_ drawable: VectorDrawable) {
+        self.drawable = drawable
+    }
+    
+    public func makeUIView(context: UIViewRepresentableContext<VectorDrawableView>) -> VectorView {
```

Comment:
shouldn't there be a body function for swiftui?

Reply:
No, this isn't a `View`. It's a `UIViewRepresentable`.

Reply:
Okay technically this is a `View`. But its `Body` is already specified as `Never` in `UIKitRepresentable,` which is a subtype of `View`.

Reply:
Wait really? `UIViewRepresentable` is a `View` as well so it can have all of the standard modifiers applied to it?